tools/hive-bigquery/bigquery_component.py

```
@@ -0,0 +1,632 @@
+            logger.debug("Deleting tracking table and BigQuery table...")
+            mysql_component.drop_table(hive_table_model.tracking_table_name)
+            self.delete_table(bq_table_model.dataset_id,
+                              bq_table_model.table_name)
+            hive_table_model.is_first_run = True
+
+        elif PropertiesReader.get('bq_table_write_mode') == "create":
+            if not hive_table_model.is_first_run:
+                print_and_log("Tracking Table {} already exist".format(
+                    hive_table_model.tracking_table_name), logging.CRITICAL)
+                return False
+            if self.check_bq_table_exists(bq_table_model.dataset_id,
+                                          bq_table_model.table_name):
+                print_and_log(
+                    "BigQuery Table {} already exist in {} dataset".format(
+                        bq_table_model.table_name, bq_table_model.dataset_id),
+                    logging.CRITICAL)
+
+                return False
+
+        else:
+            if hive_table_model.is_first_run is False:
+                query = "SELECT COUNT(*) FROM {} WHERE " \
+                        "bq_job_status='RUNNING' OR " \
+                        "bq_job_status='DONE'".format(
+                            hive_table_model.tracking_table_name)
+                results = mysql_component.execute_query(query)
+                if results[0][0] != 0:
+                    if not self.check_bq_table_exists(bq_table_model.dataset_id,
+                                                      bq_table_model.table_name):
+                        print_and_log(
+                            "Found the tracking table but BigQuery Table {} "
+                            "doesn't exist in {} dataset. Clean up the "
+                            "resources and try again",
+                            logging.CRITICAL)
+                        return False
```

Instead of returning false and logging this raise `google.api_core.exceptions.NotFound`